### PR TITLE
Add UI in option page to view and remove user rules

### PR DESCRIPTION
**Pull request by cschanaj**
@Hainish This will introduce a simple UI in the option page for users to view and remove the user rules.

P.S. to my surprise, there is some constraint in the current implementation (in lookups & caching) which rendered this more complicated than I have expected... And it is not possible to allow custom user rules names, meaning there is only 1 custom rule for each domain... 

supersedes #16616 relates #14969

### CODE REVIEW

File: chromium/pages/options/ux.js [hunk 1]
```diff
@@ -233,6 +233,40 @@ document.addEventListener("DOMContentLoaded", () => {
...
+
+      user_rule_host.className = "user-rules-list-item";
+      user_rule_name.className = "user-rules-list-item_single"
+      user_rule_name.innerText = userRule.name;
+      user_rule_host.appendChild(user_rule_name);
+      user_rules_parent.appendChild(user_rule_host);
+      user_rule_host.appendChild(remove);
+
+      remove.addEventListener("click", () => {
+        sendMessage("remove_rule", userRule, () => {
```
Comment (cschanaj) [inline chromium/pages/options/ux.js]: need fix... the second parameter should be of type `Ruleset`

File: chromium/background-scripts/store.js [hunk 2]
```diff
@@ -51,29 +51,50 @@ function local_set_promise(key, value) {
...
-      if (ls.hasOwnProperty(key)) {
-        if (key == rules.RuleSets().USER_RULE_KEY){
-          await set_promise(rules.RuleSets().USER_RULE_KEY, JSON.parse(ls[key]));
-        } else {
-          ruleActiveStates[key] = (ls[key] == "true");
+  let migration_version = await get_promise('migration_version', 0);
+
+  try {
+    if (migration_version === 0) {
+      let ls = localStorage;
```
[the remark this answers is not in the document]
Comment (cschanaj) [reply chromium/background-scripts/store.js]: resolved in 84bf30d.

**Comment by cschanaj** [comment]
@Hainish I have added new commit addresses your comments. Thanks for reviewing this!!

File: chromium/background-scripts/store.js [hunk 3]
```diff
@@ -72,29 +72,30 @@ async function performMigrations() {
...
-        })
-
-      migration_version = 2;
-      await set_promise('migration_version', migration_version);
-    }
   } catch (e) {
     // do nothing
   }
+
+  if (migration_version === 1) {
```
Comment (Hainish) [inline chromium/background-scripts/store.js]: Oh, and I guess we want this to be `migration_version < 1` since we still want to run this if it's `0`

**Comment by cschanaj** [comment]
@Hainish Travis finally passed, see 99def82. thanks!!

**Comment by cschanaj** [comment]
@Hainish Is there anything blocking this PR?

**Comment by Hainish** [comment]
Hey @cschanaj, we just had a work retreat so I'm just catching up on PRs.  I'll take a look at this tomorrow.

**Comment by Hainish** [comment]
Those changes LGTM, thanks.

**Comment by cschanaj** [comment]
thank you!

**Pull request merged**